### PR TITLE
Porting mechanics to new architecture

### PR DESCRIPTION
**Pull request by npaster**
Hi,

I have ported the mechanical module to the new architecture. 

I have also fixed some warning. I can't compile diffusion without mumps

In a soon future, I will update basis functions (to add recent works in the old basis) and quadrature for very distorted polygons

### CODE REVIEW

File: libdiskpp/include/diskpp/quadratures/bits/quad_raw_gauss.hpp
Comment (datafl4sh) [inline libdiskpp/include/diskpp/quadratures/bits/quad_raw_gauss.hpp:1]: Please restore the old formatting for this file, especially for the quadrature points data.

File: libdiskpp/include/diskpp/quadratures/bits/quad_raw_tetra.hpp [hunk 1]
```diff
@@ -10,78 +10,74 @@
...
 {
     return std::abs(v0.dot(v1.cross(v2))) / 6.;
 }
 
-static const double arbq_to_ref_A[3][3] = {
-    {  0.500000000000000, -0.288675134594813, -0.204124145231932 },
-    {  0.000000000000000,  0.577350269189626, -0.204124145231932 },
-    {  0.000000000000000,  0.000000000000000,  0.612372435695795 }
-};
+static const double arbq_to_ref_A[3][3] = {{0.500000000000000, -0.288675134594813, -0.204124145231932},
```
Comment (datafl4sh) [inline libdiskpp/include/diskpp/quadratures/bits/quad_raw_tetra.hpp:36]: Please restore old formatting for `arbq_to_ref_A`.

**Review by datafl4sh — CHANGES_REQUESTED** [review]
Hello @npaster . I reviewed all the changes, thank you for your huge work. I have two major remarks. The first is to please do not use tools that automatically reformat the code: they add a lot of noise, and reviewing the changes becomes very hard. It is easy to miss some bugs in the middle of all the formatting changes. The second is that some functions were removed from `hho_degree_info` and this breaks lots of stuff. Please be sure to do a full compile.

**Comment by npaster** [comment]
> Hello @npaster . I noticed that you sent a PR, but in the last months I was really busy...teaching and stuff...so I wasn't working on code at all. Sorry for the delay, and thank you for your work. I am going to merge it.

Hello Matteo,

Thank you for your review.

I am going to modify my code next week. I will try to remove my automatic reformatting

**Comment by npaster** [comment]
I hope that everything is fixed.

My compiler is not fully c++20 compatible and I have many errors

I have also a lot with mumps (I don't have mumps locally)

/diskpp/apps/diffusion/src/diffusion_vem.cpp:138:12: error: use of undeclared identifier 'mumps_lu'
    csol = mumps_lu(LHS, RHS);

**Pull request merged**